app/src/androidTest/java/ch/epfl/sweng/runpharaa/TrackPropertiesActivityTest.java

```diff
@@ -201,6 +199,25 @@ public void correctValuesDisplayedForTrack1() {
         withId(R.id.trackLikesID).matches(withText("Likes: " + t1.getProperties().getLikes()));
     }
 
+    @Test
+    public void deletedTrackTest(){
+        Track t1 = createTrack();
+        String s = User.instance.getUid();
+        User.instance.setUid("BobUID");
+        launchWithExtras(t1);
+        onView(withId(R.id.deleteButton))
+                .perform(click());
+        onView(withText(R.string.cancel)).perform(click());
+
+        onView(withId(R.id.deleteButton))
+                .perform(click());
+        onView(withText(R.string.delete)).perform(click());
+
+        assertTrue(mActivityRule.getActivity().isFinishing());
+
+        User.instance.setUid(s);
```

Comment:
Why do you do that after the assert?

Reply:
Because it needed to launch trackPorpertiesActivity on a track taht can be deleted(User.instance one). It was not the case with the default database mock. So i switch the use uid only for this test and put it back to not mess up with other tests